packages/editor/src/client/api/setting.graphql

```diff
@@ -0,0 +1,44 @@
+fragment FullSetting on Setting {
+    id
+    name
+    value
+    settingRestriction {
+        maxValue
+        minValue
+        inputLength
+        allowedValues
+    }
+}
+
+query SettingList {
+  settings {
+    id
+    name
+    value
+    settingRestriction {
+        maxValue
+        minValue
+        inputLength
+        allowedValues
+    }
+  }
+}
+
+query Setting($id: ID, $name: String) {
+    setting(id: $id, name: $name) {
+        id
+        value
+        settingRestriction {
+            maxValue
+            minValue
+            inputLength
+            allowedValues
+        }
+    }
+}
+
+mutation UpdateSetting($id: String!, $input: SettingInput!) {
+  updateSetting(id: $id, input: $input) {
+    value
+  }
+}
```

Comment:
Its for the frontend access to the graphql so I haven't had use for it. In fact I'm not sure if its necessary at all because settings would need a migration to add or remove. Maybe we can discuss if this is the ideal approach.

Reply:
Yeah imo settings should only be deleted/added via migrations